mkosi

```diff
@@ -761,6 +761,23 @@ def install_distribution(args, workspace, run_build_script):
 
     install[args.distribution](args, workspace, run_build_script)
 
+def reset_machine_id(args, workspace):
+
+    # Make /etc/machine-id an empty file. This way, on the next boot
+    # is either initialized and commited (if /etc is writable) or the
+    # image runs with a transient machine ID, that changes on each
+    # boot (if the image is read-only).
+
+    print_step("Resetting machine ID...")
+
+    machine_id = os.path.join(workspace, 'root', 'etc/machine-id')
+
+    os.unlink(machine_id)
+    f = open(machine_id, "w+b");
```

Comment:
I'd prefer `open(machine_id, "w+b").close()`.

(While it's true that with cpython, `del` causes `close` to be called immediately, but in principle somebody could use `pypy3` to run this, at least once it starts supporting modern python syntax again. So, let's keep up the illusion of portability and not require destructors to be called immediately ;))